desci-server/src/controllers/proxy/orcidProfile.ts

```diff
@@ -0,0 +1,93 @@
+import axios from 'axios';
+import { Request, Response, NextFunction } from 'express';
+
+import prisma from 'client';
+import parentLogger from 'logger';
+
+export const ORCID_API_URL = 'https://orcid.org';
+
+interface OrcidQueryResponse {
+  ok: boolean;
+  profile: any;
+}
+interface OrcidQueryResponseError {
+  ok: boolean;
+  error: string;
+}
+
+const ORCID_PROFILE_TTL = 1000 * 60 * 60 * 24; // 1 day
+const REFRESH_RATE_LIMIT = 1000 * 60 * 5; // 5 minutes
+
+export const orcidProfile = async (
+  req: Request,
+  res: Response<OrcidQueryResponse | OrcidQueryResponseError>,
+  next: NextFunction,
+) => {
+  debugger;
+  const orcidId = req.params.orcidId as string;
+
+  const { refresh: refreshQuery } = req.query;
+  const refresh = refreshQuery === 'true';
+
+  if (!orcidId) return res.status(400).json({ ok: false, error: 'orcidId required' });
+
+  const logger = parentLogger.child({
+    module: 'SERVICES::orcidQueryController',
+    orcidId,
+    refresh,
+  });
+
+  try {
+    // fetch from cache
+    const cachedResult = await prisma.orcidProfile.findFirst({ where: { orcidId: orcidId } });
+    // check ttl against last updated
+    const isFresh = cachedResult?.updatedAt < cachedResult?.expiresIn;
+
+    if (cachedResult && isFresh && !refresh) {
+      logger.info(
+        { updatedAt: cachedResult.updatedAt, expiresIn: cachedResult.expiresIn },
+        'Cached result found, returning',
+      );
+      return res.status(200).send({ ok: true, profile: cachedResult.profile });
+    } else {
+      //rate limit on refresh
+      if (cachedResult && refresh) {
+        const rateLimitDate = new Date(Date.now() - REFRESH_RATE_LIMIT);
+        const isRecentlyRefreshed = cachedResult.updatedAt > rateLimitDate;
+        if (isRecentlyRefreshed) {
```

Comment:
would prefer returning a successful result and using the "refresh" param as a suggestion vs 429. Two users independently of each other could call refresh=true and one would get an error and not know why